### PR TITLE
replace the crate jdbc with the pgjdbc driver

### PR DESCRIPTION
**Pull request by kovrus**
**DO NOT MERGE! WAIT FOR 0.56.0!**

### CODE REVIEW

File: java/README.md [hunk 1]
```diff
@@ -1,34 +1,61 @@
...
+- [PostgreSQL JDBC][2]
 - [Java 8][3]
 
 ## Usage
-To build and run the sample application we use the [Apache Maven][4] build tool.
+To build and run the sample application we use the [Apache Maven][4]
+build tool.
+
+### Run Crate Server
+Get the latest Crate [nightly build][5] and start the Crate from the
```
Comment (chaudum) [inline java/README.md]: This demo requires the experimental Postgres Protocol for Crate which will be releases with the `0.56.0`. Until there is no such release, you need to us the latest nightly build ....

File: java/README.md [hunk 2]
```diff
@@ -1,34 +1,61 @@
...
+tarball. The postgres wire protocol is disabled by default. Therefore,
+please make sure that it is enabled, by setting it to _true_ in the
```
Comment (chaudum) [inline java/README.md]: formatting: code instead of italic

Comment (chaudum) [inline java/README.md]: check all occurrences of `_***_` with "`"

**Comment by chaudum** [comment]
otherwise LGTM
maybe @mikethebeer also wants to take a look.

this PR should not be merged until the new Postgres JDBC driver is available.

**Comment by mikethebeer** [comment]
LGTM 👍

File: java/src/main/java/io/crate/jdbc/sample/Controller.java [hunk 3]
```diff
@@ -27,14 +27,30 @@
     private static final int CREATED = 201;
     private static final int OK = 200;
 
-    public Controller(final DataProvider model) {
+    Controller(final DataProvider model) {
 
         before(((request, response) -> {
             response.header("Access-Control-Allow-Origin", "*");
             response.header("Access-Control-Request-Method", "*");
```
Comment (chaudum) [inline java/src/main/java/io/crate/jdbc/sample/Controller.java]: this header required? seems like it was a wrong header and the new one is superseding that

File: java/src/main/java/io/crate/jdbc/sample/Controller.java [hunk 4]
```diff
@@ -27,14 +27,30 @@
...
+            response.header("Access-Control-Allow-Methods", "*");
             response.header("Access-Control-Allow-Headers", "*");
         }));
 
+        options("/*", (request, response) -> {
```
Comment (chaudum) [inline java/src/main/java/io/crate/jdbc/sample/Controller.java]: don't quite understand this behaviour.
why are certain headers checked for existence, others not?

File: java/pom.xml [hunk 5]
```diff
@@ -63,6 +56,18 @@
             <artifactId>httpclient</artifactId>
             <version>${apache.httpcomponents.version}</version>
         </dependency>
+
+        <dependency>
```
Comment (chaudum) [inline java/pom.xml]: why is guava a new dependency?

Comment (kovrus) [inline java/pom.xml]: yes it is the new one, it was a transitive dependency of crate-client before

**Pull request merged**